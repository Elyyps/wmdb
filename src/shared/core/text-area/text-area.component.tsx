import * as React from "react";
import styles from "./text-area-component.module.scss";
import classNames from "classnames";
import Cross from "@assets/icons/cross.svg";
import Check from "@assets/icons/check.svg";
import { IconComponent } from "../icon";

export interface ITextAreaComponentProps {
  classModify?: string;
  errorMessage?: string;
  icon?: string;
  isError?: string;
  isSuccess?: boolean;
  label?: string;
  name: string;
  placeholder?: string;
}

const TextAreaComponent = (props: ITextAreaComponentProps) => {
  const { name, placeholder, classModify, icon, label, isError, isSuccess } = props;
  const inputClassName = classNames("textarea", {
    [`textarea--${classModify}`]: classModify
  });
  const [value, setValue] = React.useState("");
  const wrapperClassnames = classNames({
    ["error"]: isError,
    ["isIcon"]: icon,
    ["success"]: isSuccess
  });
  const handelChange = (event: any) => {
    {
      setValue(event.target.value);
    }
  };

  const handelClear = () => {
    setValue("");
  };

  return (
    <div className={styles["text-area"]}>
      <React.Fragment>
        <div className={` ${styles["form__item"]} ${wrapperClassnames}`}>
          {label && (
            <label htmlFor="label" className={styles["form__item-label"]}>
              {label}
            </label>
          )}
          <div className={styles["form__item-holder"]}>
            <textarea
              onChange={handelChange}
              placeholder={placeholder}
              className={inputClassName}
              name={name}
              value={value}
            />
            {value && !isSuccess && (
              <div role={"button"} className={styles["form__item-icon"]} onClick={handelClear}>
                <IconComponent icon={Cross} size={"10px"} />
              </div>
            )}
            {isSuccess && (
              <div className={styles["form__item-icon"]}>
                <IconComponent icon={Check} size={"10px"} />
              </div>
            )}
          </div>

          {isError && <div className={styles["error-message"]}>{isError && isError}</div>}
        </div>
      </React.Fragment>
    </div>
  );
};

export { TextAreaComponent };
