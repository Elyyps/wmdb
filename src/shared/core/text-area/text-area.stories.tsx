import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { TextAreaComponent } from "./text-area.component";

storiesOf("TextArea", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <TextAreaComponent name={"name"} />);
