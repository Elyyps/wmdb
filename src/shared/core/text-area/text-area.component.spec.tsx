import React from "react";
import { shallow } from "enzyme";
import { TextAreaComponent } from "./text-area.component";

describe("[TextArea]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<TextAreaComponent name={"name"} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
