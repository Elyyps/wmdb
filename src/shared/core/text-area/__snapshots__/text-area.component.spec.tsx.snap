// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TextArea] should render component without crashing 1`] = `
<div
  className="text-area"
>
  <div
    className=" form__item "
  >
    <div
      className="form__item-holder"
    >
      <textarea
        className="textarea"
        name="name"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
</div>
`;
